api/guids/views.py

```diff
@@ -33,10 +35,29 @@ class GuidRedirect(JSONAPIBaseView):
     view_category = 'guids'
     view_name = 'guid-detail'
 
+    serializer_class = GuidSerializer
+
+    @staticmethod
+    def should_resolve(request):
+        resolve = request.query_params.get('resolve')
+        return resolve is None or is_truthy(resolve)
```

Comment:
Oh, I think I was misunderstanding the meaning of resolve. 

"resolve" = redirect to the referent's canonical URL

OK, this is good.